src/node_contextify.cc

```diff
@@ -934,6 +934,30 @@ Maybe<bool> StoreCodeCacheResult(
   return Just(true);
 }
 
+// TODO(RaisinTen): Reuse in ContextifyContext::CompileFunction().
+MaybeLocal<Function> CompileFunction(Isolate* isolate,
+                                     Local<Context> context,
+                                     Local<String> filename,
+                                     Local<String> content,
+                                     std::vector<Local<String>>* parameters) {
+  ScriptOrigin script_origin(isolate, filename, 0, 0, true);
+  ScriptCompiler::Source script_source(content, script_origin);
+
+  Local<Function> fn;
+  if (!ScriptCompiler::CompileFunction(context,
+                                       &script_source,
+                                       parameters->size(),
+                                       parameters->data(),
+                                       0,
+                                       nullptr,
+                                       ScriptCompiler::kEagerCompile)
```

Comment:
Note I recently fixed a bug where ScriptCompiler::CompileFunction ignores kEagerCompile ([v8 cl](https://chromium-review.googlesource.com/c/v8/v8/+/4571822)). I wonder if you apply that patch locally, you might see better results?

Reply:
Amazing, I was able to get a 33% improvement using your patch! Backporting it in https://github.com/nodejs/node/pull/48576.